man/adoc/bpftrace.adoc

```diff
@@ -1633,6 +1633,16 @@ bpftrace doesn't read past the length of the shortest string.
 
 The use of the `==` and `!=` operators is recommended over calling `strncmp` directly.
 
+=== strstr
+
+.variants
+* `int64 strstr(char * s1, char * s2)`
+
+`strstr` compares whether the string val1 contains the string val2.
+If s2 is contained `0` is returned, else a non-zero value is returned.
```

Comment:
I feel like this was discussed before but I'm not convinced by the `strstr` name.  This function doesn't actually behave the same as `strstr` does in the C libs.

Maybe `strcontains` is a better name?

From the `strstr` manpages:

>      If needle is an empty string, haystack is returned; if needle occurs nowhere in haystack, NULL is returned; otherwise a pointer to the first character of the first occurrence of needle is returned.

This one behaves completely different, it returns 0 *when found*.

Reply:
@BurntBrunch @viktormalik  how do you feel about this rename? it got implemented now but I wanted to start a discussion :)

Reply:
I think that it's fine, although I'd expect it to return 1 if the substring is found, so that you can test for the positive case instead of the negative one when using it in a condition.

Another option would be to introduce a new operator for this (`in`?) but that's a bigger change and would have to be discussed more, I guess.

Reply:
Ah, sorry, I prefer `strcontains` since, as you said, `strstr` has a different semantics in C.